my-app/src/app/app.module.ts

```diff
@@ -5,13 +5,23 @@ import { AppRoutingModule } from './app-routing.module';
 import { AppComponent } from './app.component';
 import { MainLayoutModule } from './main-layout/main-layout.module';
 import { PageNotFoundComponent } from './page-not-found/page-not-found.component';
+import { StoreModule } from '@ngrx/store';
+import { reducers } from './core/store/app.state';
+import { environment } from 'src/environments/environment';
+import { StoreDevtoolsModule } from '@ngrx/store-devtools';
+import { EffectsModule } from '@ngrx/effects';
+import { AuthEffects } from './core/store/auth/effects/auth.effects';
+import { CoursesEffects } from './core/store/courses/effects/course.effects';
```

Comment:
in each folder should be an index.ts file which exports everything from the folder.
So, your import should look like from './core/index'